matlab/src/cpp/arrow/matlab/type/proxy/field.cc

```diff
@@ -0,0 +1,129 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow/util/utf8.h"
+
+#include "arrow/matlab/type/proxy/field.h"
+#include "arrow/matlab/error/error.h"
+
+#include "arrow/matlab/type/proxy/primitive_ctype.h"
+#include "arrow/matlab/type/proxy/timestamp_type.h"
+#include "arrow/matlab/type/proxy/string_type.h"
+
+#include "libmexclass/proxy/ProxyManager.h"
+
+namespace arrow::matlab::type::proxy {
+
+    Field::Field(std::shared_ptr<arrow::Field> field) : field{std::move(field)} {
+        REGISTER_METHOD(Field, name);
+        REGISTER_METHOD(Field, type);
+        REGISTER_METHOD(Field, toString);
+    }
+
+    std::shared_ptr<arrow::Field> Field::unwrap() {
+        return field;
+    }
+
+    void Field::name(libmexclass::proxy::method::Context& context) {
+        namespace mda = ::matlab::data;
+        mda::ArrayFactory factory;
+
+        const auto str_utf8 = field->name();
```

Comment:
Is there a way to get a reference from `field`? If so, we could avoid a copy. Not a big deal though.

Reply:
That's a good point. I will take a look at the `field` APIs to see if there is anyway to get a reference or `string_view` instead.

Reply:
It turns out that `name` actually does return a `const std::string&`. I just didn't include the `&` on the output variable type. Fixed.